### PR TITLE
Added the basic_auth_disabled decorator

### LINKED ISSUE #10: apply basic auth to all endpoints except for one (or excluded pattern)
Hi.
Thanks for this package. I am using it to apply a basic auth restriction to staging deploys for a couple of my websites.

I host my sites on AWS behind an Elastic Load Balancer (ELB). When you set up an ELB, you give it a 'health check' URL. The Load Balancer periodically requests this URL and checks the responses status is `200 OK` to ensure the server is 'healthy'. If it gets anything other than a `200 OK`, it marks the server as unhealthy. The load balancer doesn't have the option to pass auth credentials.

What this means is I want to set up my staging deploys with basic auth applied to every endpoint _except for one_ which isn't protected so that the load balancer can ping that endpoint and get a `200 OK` rather than a `401 Unauthorized`. Every other route should be behind basic auth. For obvious reasons, it is cumbersome to decorate every view other than one with `@basic_auth_required` so they way I've done this is by subclassing `BasicAuthMiddleware` so that I can turn on basic auth for everything but then whitelist one endpoint using a regex. I've taken inspiration from the old [django-moat](https://github.com/amrox/django-moat) package which is no longer maintained but did have this feature.
example: https://github.com/DemocracyClub/EveryElection/blob/master/every_election/apps/core/middleware.py

2 questions:

1. Is there a better way of doing this that I am missing?
2. If not, would you accept a PR to add this feature to the package?

Cheers

### PR DESCRIPTION
**Pull request by fleaz**
Hey,
I implemented another decorator which can be used to disable BasicAuth on a single view basis if it has been enabled globaly via the middleware integration. This fixes #10 

Best,
Felix

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -73,6 +73,24 @@ Authenticated users can pass it without `Basic ...` header.
 ``target_test`` accepts ``typing.Callable[[HttpRequest], bool]``,
 and if the callable returns ``True``, Basic Auth will be required.
 
+Disable Basic Auth for specific requests
+-------------------------------------
```
Comment (gmseb) [inline README.rst:77]: The `----` bar seems to be three characters short, please add three more dashes.

File: basicauth/decorators.py [hunk 2]
```diff
@@ -4,14 +4,21 @@
 from .basicauthutils import validate_request
 
 
-def basic_auth_required(func=None,
-                        target_test=(lambda request: True)):
+def basic_auth_disabled(func=None):
+    def _wrapped(request, *args, **kwargs):
+        return func(request, *args, **kwargs)
```
Comment (gmseb) [inline basicauth/decorators.py:9]: With this wrapper…
- it is not clear to me how this would talk the middleware out of enforcing basicauth
- this seems to be effectively `_wrapped = func`, which would seem pointless to me
- if this keeps using some wrapper, the wrapper should probably be decorated using `@wraps`

PS: This hack is probably too dirty, but it should work:
```python
from django.test import override_settings

with override_settings(BASICAUTH_DISABLE=True):
    ...
```

File: basicauth/decorators.py [hunk 3]
```diff
@@ -4,14 +4,21 @@
...
+
+
+def basic_auth_required(func=None, target_test=(lambda request: True)):
     def actual_decorator(view_func):
         @wraps(view_func)
         def _wrapped(request, *args, **kwargs):
             if target_test(request) and not validate_request(request):
                 return HttpResponseUnauthorized()
             return view_func(request, *args, **kwargs)
+
```
Comment (gmseb) [inline basicauth/decorators.py:21]: The addition in whitespace seems unrelated, my vote for a revert.

**Review by gmseb — CHANGES_REQUESTED** [review]
@fleaz this will likely not work as-is, please see my comments below.

@hirokiky is there still interest in a working implementation to this feature?

**Comment by fleaz** [comment]
Thanks for your review!

But this PR has been sitting idle without any interaction for three years now.
In the meantime I switched jobs, and haven't touched any Django code for multiple years.
Also this repo appears to be completely dead for even longer than this PR has been stale.

I'm sorry, but I feel like spending time to getting back into Django and working on this PR is wasted.

**Comment by gmseb** [comment]
@fleaz that's fair, thanks for your quick reply.
